### PR TITLE
Doc improvement

### PR DESCRIPTION
**Pull request by vincent-unity**
Updated the document by:
1. Split the one doucment to several md files to make the doc better organized.
2. Added missing document and update some unclear document.

To QA:
Please verify the document by below steps:
1. Clone this repo https://github.cds.internal.unity3d.com/unity/com.unity.package-manager-doctools/
2. Add the cloned package to your project through Package Manager.
3. Restart Package Manager, there will be a ‘Generate Documentation’ button shown under your package page.

Please make sure all the links in the document are correct. I personally verified them, but would like you can do a double check.

### CODE REVIEW

File: com.unity.mobile.notifications/Documentation~/iOS.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    // Second = 0
+    Repeats = false
+};
+```
+
+#### Location trigger
+
+You can also create an [iOSNotificationLocationTrigger](../api/Unity.Notifications.iOS.iOSNotificationLocationTrigger.html) if you want to schedule the delivery of a notification when the device enters or leaves a specific geographic region.
+
+Before you schedule any notifications with this trigger, you need to check `Include CoreLocation Framework` in the [notifications settings](settings.html#include-corelocation-framework). Your app must have authorization to use Core Location and must have when-in-use permissions. You can use the Unity LocationService API to request this authorization.
```
Comment (keiramoon) [inline com.unity.mobile.notifications/Documentation~/iOS.md:115]:
```suggestion
Before you schedule any notifications with this trigger, you need to enable the **Include CoreLocation Framework** option in the [notifications settings](settings.html#include-corelocation-framework). Your app must have authorization to use Core Location and must have when-in-use permissions. You can use the Unity LocationService API to request this authorization. 
```

File: com.unity.mobile.notifications/Documentation~/iOS.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+```
+
+#### Location trigger
+
+You can also create an [iOSNotificationLocationTrigger](../api/Unity.Notifications.iOS.iOSNotificationLocationTrigger.html) if you want to schedule the delivery of a notification when the device enters or leaves a specific geographic region.
+
+Before you schedule any notifications with this trigger, you need to check `Include CoreLocation Framework` in the [notifications settings](settings.html#include-corelocation-framework). Your app must have authorization to use Core Location and must have when-in-use permissions. You can use the Unity LocationService API to request this authorization. 
+For additional information, see the [Core Location](https://developer.apple.com/documentation/corelocation/clregion?language=objc) documentation on the Apple Developer website.
+
+In the below example, the center coordinate is defined using the WGS 84 system. The app triggers the notification when the user enters an area within a 250 meter radius around the Eiffel Tower in Paris.
```
Comment (keiramoon) [inline com.unity.mobile.notifications/Documentation~/iOS.md:118]:
```suggestion
In the example below, the center coordinate is defined using the WGS 84 system. The app triggers the notification when the user enters an area within a 250 meter radius around the Eiffel Tower in Paris.
```

File: com.unity.mobile.notifications/Documentation~/iOS.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+    Center = new Vector2(2.294498f, 48.858263f),
+    Radius = 250f,
+    NotifyOnEntry = true,
+    NotifyOnExit = false,
+}
+```
+
+#### Push trigger
+
+You shouldn't really create an [iOSNotificationPushTrigger](../api/Unity.Notifications.iOS.iOSNotificationPushTrigger.html) instance as you can't really send a push notification from your app. This notification trigger is used to indicate if the notification was sent from Apple Push Notification Service (APNs).
```
Comment (keiramoon) [inline com.unity.mobile.notifications/Documentation~/iOS.md:132]:
```suggestion
You shouldn't need to create an [iOSNotificationPushTrigger](../api/Unity.Notifications.iOS.iOSNotificationPushTrigger.html) instance as you can't really send a push notification from your app. This notification trigger is used to indicate if the notification was sent from Apple Push Notification Service (APNs).
```

File: com.unity.mobile.notifications/Documentation~/iOS.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+
+### Notification received callbacks
+
+#### [iOSNotificationCenter.OnNotificationReceived](../api/Unity.Notifications.iOS.iOSNotificationCenter.html#Unity_Notifications_iOS_iOSNotificationCenter_OnNotificationReceived)
+
+By default if your app triggers a local notification while it is in the foreground, the device won’t display an alert for that notification. If you want the notification to behave as though the device isn’t running the app, set the `ShowInForeground` property when you schedule the notification as below.
```
Comment (keiramoon) [inline com.unity.mobile.notifications/Documentation~/iOS.md:138]:
```suggestion
By default, if your app triggers a local notification while it is in the foreground, the device won’t display an alert for that notification. If you want the notification to behave as though the device isn’t running the app, set the `ShowInForeground` property to true when you schedule the notification, as shown below.
```

File: com.unity.mobile.notifications/Documentation~/iOS.md [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+
+// In this case you need to specify its 'ForegroundPresentationOption'
+notification.ForegroundPresentationOption = (PresentationOption.Sound | PresentationOption.Alert);
+```
+
+By subscribing to the `iOSNotificationCenter.OnNotificationReceived` event, you can also perform other actions when the app triggers a notification. For example, you can display the notification content using the app’s UI. Your app calls this event whenever a local or a remote notification is received, regardless if it's shown in the foreground.
+
+#### [iOSNotificationCenter.OnRemoteNotificationReceived](../api/Unity.Notifications.iOS.iOSNotificationCenter.html#Unity_Notifications_iOS_iOSNotificationCenter_OnRemoteNotificationReceived)
+
+To modify or hide the content of a received remote notification while your app is running, subscribe to the `iOSNotificationCenter.OnRemoteNotificationReceived` event. With subscribing to this event, the remote notification won’t display when your app is running. If you still want to show an alert for it, schedule a local notification using the remote notification’s content. Below is an example of how to do this.
```
Comment (keiramoon) [inline com.unity.mobile.notifications/Documentation~/iOS.md:151]:
```suggestion
To modify or hide the content of a received remote notification while your app is running, subscribe to the `iOSNotificationCenter.OnRemoteNotificationReceived` event. When subscribing to this event, the remote notification won’t display when your app is running. If you still want to show an alert for it, schedule a local notification using the remote notification’s content. The example below shows how to do this.
```

File: com.unity.mobile.notifications/Documentation~/iOS.md [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+    msg += "\n - .Badge: " + notification.Badge;
+    msg += "\n - .Body: " + notification.Body;
+    msg += "\n - .CategoryIdentifier: " + notification.CategoryIdentifier;
+    msg += "\n - .Subtitle: " + notification.Subtitle;
+    msg += "\n - .Data: " + notification.Data;
+    Debug.Log(msg);
+}
+```
+
+If the user opens the app from a notification, [iOSNotificationCenter.GetLastRespondedNotification](../api/Unity.Notifications.iOS.iOSNotificationCenter.html#Unity_Notifications_iOS_iOSNotificationCenter_GetLastRespondedNotification) also returns that notification. Otherwise it returns null.
```
Comment (keiramoon) [inline com.unity.mobile.notifications/Documentation~/iOS.md:206]:
```suggestion
If the user opens the app from a notification, [iOSNotificationCenter.GetLastRespondedNotification](../api/Unity.Notifications.iOS.iOSNotificationCenter.html#Unity_Notifications_iOS_iOSNotificationCenter_GetLastRespondedNotification) also returns that notification. Otherwise, it returns null.
```

File: com.unity.mobile.notifications/Documentation~/iOS.md [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+
+#### Set custom data for remote notifications
+
+Sometimes users might want to set custom data on the payload for a remote notification and retrieve it by [iOSNotification.Data](../api/Unity.Notifications.iOS.iOSNotification.html#Unity_Notifications_iOS_iOSNotification_Data). Below is an example of setting a string as `data` on the payload.
```
Comment (keiramoon) [inline com.unity.mobile.notifications/Documentation~/iOS.md:210]:
```suggestion
Sometimes, you might want to set custom data on the payload for a remote notification and retrieve it using [iOSNotification.Data](../api/Unity.Notifications.iOS.iOSNotification.html#Unity_Notifications_iOS_iOSNotification_Data). The example below shows how to set a string as `data` on the payload.
```

Comment (keiramoon) [reply com.unity.mobile.notifications/Documentation~/iOS.md:210]: I assumed "users" here refers to developers, i.e. people who use this package, in which case we need to write this in second-person ("you"). "Users" in the context of these package docs refers to end-users.

File: com.unity.mobile.notifications/Documentation~/iOS.md [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+        "alert": {
+            "title": "Hello world!",
+            "body": "This is an example of a remote notification"
+            }
+    },
+    "data": "Test data"
+}
+```
+
+You have to use the exact `data` as the key of your custom data as it's what the package looks for.
```
Comment (keiramoon) [inline com.unity.mobile.notifications/Documentation~/iOS.md:224]:
```suggestion
You have to use the exact `data` as the key of your custom data, because this is what the package looks for.
```

File: com.unity.mobile.notifications/Documentation~/index.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
+# Introduction
+
+Mobile Notification package adds support for scheduling local one-time or repeatable notifications on Android and iOS platforms.
```
Comment (keiramoon) [inline com.unity.mobile.notifications/Documentation~/index.md:3]:
```suggestion
The Unity Mobile Notifications package adds support for scheduling local one-time or repeatable notifications on Android and iOS.
```

File: com.unity.mobile.notifications/Documentation~/index.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+### Requirements
+
+- Compatible with Unity 2018.3 or above.
+- Compatible with Android 4.4+ (API 19) and iOS 10.0+.
+
+### Supported features
+
+The runtime notification APIs are split into two parts for both Android and iOS. These APIs can be used to schedule and manage notifications as listed blow:
```
Comment (keiramoon) [inline com.unity.mobile.notifications/Documentation~/index.md:12]:
```suggestion
The runtime notification APIs are split into two parts for both Android and iOS. These APIs can be used to schedule and manage notifications as listed below:
```

File: com.unity.mobile.notifications/Documentation~/index.md [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+
+*   Schedule local one-time or repeatable notifications.
+*   Cancel already displayed and upcoming (scheduled) notifications.
+*   Android:
+    *   Create and modify notification channels (categories) on Android Oreo and above.
```
Comment (keiramoon) [inline com.unity.mobile.notifications/Documentation~/index.md:17]:
```suggestion
    *   Create and modify notification channels (categories) on Android 8.0 (Oreo) and above.
```

Comment (keiramoon) [reply com.unity.mobile.notifications/Documentation~/index.md:17]: Just to make this easier to understand for people who might not be as familiar with Android codenames. Do we need to call out an Android API level here as well?

File: com.unity.mobile.notifications/Documentation~/Android.md [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+AndroidNotificationCenter.RegisterNotificationChannel(channel);
+```
+
+You can check what other properties you can set at [AndroidNotificationChannel](../api/Unity.Notifications.Android.AndroidNotificationChannel.html).
+
+You can also delete or get a notification channel, etc. Please refer to [AndroidNotificationCenter](../api/Unity.Notifications.Android.AndroidNotificationCenter.html) for more notification channel related APIs.
+
+Once you create a notification channel, you cannot change the behavior of it, read more at [Android Notification Channel Document](https://developer.android.com/training/notify-user/channels).
+
+On devices which are lower than Android 8.0, this package emulates the same behavior by applying properties on notification channels like `Importance` to individual notifications.
```
[the remark this answers is not in the document]
Comment (vincent-unity) [reply com.unity.mobile.notifications/Documentation~/Android.md:24]: There are no notification channels on Android 7 or below, but our APIs design still follow the one on Android 8 and above. So we apply the properties on notification channels directly to the notifications.

File: com.unity.mobile.notifications/Documentation~/FAQ.md [hunk 13]
```diff
@@ -0,0 +1,19 @@
+# FAQ
+
+#### Why are small Android icons white in the editor notification settings preview
+
+Small notification icons are [required](https://material.io/design/platform-guidance/android-notifications.html#anatomy-of-a-notification) to be monochrome and Android ignores all non-alpha channels in the icon image, therefore Unity automatically strips all RGB channels.
+
+You can also provide icons by putting them in the `\Assets\Plugins\Android\res\drawable-{scaleFactor}`folder which will not be automatically processed, however icons which contain non alpha channel will not be correctly displayed on Android 5.0 and above.
```
[the remark this answers is not in the document]
Comment (vincent-unity) [reply com.unity.mobile.notifications/Documentation~/FAQ.md:7]: Copied from the old document. 
I believe it means our build build pipeline and the notification package post processor won't process them.

File: com.unity.mobile.notifications/Documentation~/FAQ.md [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+
+The notification color can be modified by [AndroidNotification.Color](../api/Unity.Notifications.Android.AndroidNotification.html#Unity_Notifications_Android_AndroidNotification_Color) property.
+
+#### Why are Notifications not delivered on certain Huawei and Xiaomi phones when my app is closed and not running in the background
+
+Seems that Huawei (including Honor) and Xiaomi utilize [aggresive batter saver techniques](https://stackoverflow.com/questions/47145722/how-to-deal-with-huaweis-and-xiaomis-battery-optimizations) which restricts app background activities, unless the app has been whitelisted by the user in device settings.
```
[the remark this answers is not in the document]
Comment (vincent-unity) [reply com.unity.mobile.notifications/Documentation~/FAQ.md:13]: Again, copied from the old document, will try to search around.

File: com.unity.mobile.notifications/Documentation~/Settings.md [hunk 15]
```diff
@@ -0,0 +1,35 @@
+# Notification Settings
+
+You can access `Notification settings` by menu `Edit -> Project Settings -> Mobile Notifications`, where you can control the behavior of the notification package to some extent.
```
[the remark this answers is not in the document]
Comment (vincent-unity) [reply com.unity.mobile.notifications/Documentation~/Settings.md:3]: Yep, it's confusing, we should remove it.

File: com.unity.mobile.notifications/Documentation~/iOS.md (hunk 7 above)
Comment (vincent-unity) [reply com.unity.mobile.notifications/Documentation~/iOS.md]: > I assumed "users" here refers to developers

File: com.unity.mobile.notifications/Documentation~/Android.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (keiramoon) [reply com.unity.mobile.notifications/Documentation~/Android.md]: I see! In that case, can we rephrase this to:

> On devices that use Android versions prior to 8.0, this package emulates the same behavior by applying notification channel properties, such as `Importance`, to individual notifications.

[the remark this answers is not in the document]
Comment (vincent-unity) [reply com.unity.mobile.notifications/Documentation~/Android.md]: Fixed at https://github.com/Unity-Technologies/com.unity.mobile.notifications/pull/52/commits/2699e9e3f8ea8dddb33dc9f9c66f238a587ea54f.

**Comment by vaidasma** [comment]
There' s a typo in word **aggresive** (should be aggressive). Other than that, looks good
